### PR TITLE
Enhance Layer border radius and allow background prop to be applied

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Allows user to style Layer border radius via grommet size such as "small" or px value such as "4px". If user opts into `intelligentRounding`, then the Layer will automatically apply a border-radius of 0 to any edge of the layer that's touching the edge of the screen.

For the positions "start" and "end" that are dependent on if a user is in LTR or RTL, there is a new CSS style that can be used to apply border-radius properly depending on which direction the app is in. However, it currently only functions in Firefox. So, we need to decide what the fallback behavior should be for all other browsers.

Additionally, added background prop to Layer so that user could style background and maintain border-radius theming.

#### Where should the reviewer start?
src/js/components/Layer/StyledLayer.js

#### What testing has been done on this PR?
Added Border Radius storybook example to Layer and added jest tests for all positions/full combinations.

#### How should this be manually tested?
Look at border radius storybook example.

#### Any background context you want to provide?
Layer would always round all corners. This is not always desirable for Layers with position "right" where one edge is flush against the edge of the window.

#### What are the relevant issues?
Related to https://github.com/grommet/hpe-design-system/issues/1341

#### Do the grommet docs need to be updated?
Yes, updated here.

#### Should this PR be mentioned in the release notes?
Yes. Enhance Layer to accept background prop and grommet sizes for theme.layer.border.radius.
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Layer/StyledLayer.js [hunk 1]
```diff
@@ -651,8 +742,15 @@ const desktopContainerStyle = css`
       props.theme,
       'right',
     )}px)`};
-  border-radius: ${props =>
-    props.plain ? 0 : props.theme.layer.border.radius};
+  ${props =>
+    props.plain || props.full === true
```
Comment (ericsoderberghp) [inline src/js/components/Layer/StyledLayer.js:746]: Can we have `full` with `margin`? If so, then we should use `roundStyle()` in that case too.

File: src/js/components/Layer/doc.js [hunk 2]
```diff
@@ -146,6 +148,18 @@ export const themeDoc = {
...
+    description: 'The rounding of the Layer corners.',
+    type: 'string',
+    defaultValue: 'white',
+  },
+  'layer.border.intelligentRounding': {
+    description: `Whether the border-radius of the Layer should adapt based on 
+    the Layer's position. Wherever the Layer is touching the edge of the 
+    screen, a border-radius of 0 will be applied.`,
+    type: 'boolean',
+    defaultValue: undefined,
```
Comment (ericsoderberghp) [inline src/js/components/Layer/doc.js:161]: Given that `layer.border.radius` is new, should the default value for this key should be `true`, given that's the more likely desire?

Comment (taysea) [reply src/js/components/Layer/doc.js:161]: layer.border.radius is not new, it just wasn't documented prior

File: src/js/components/Layer/StyledLayer.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Layer/StyledLayer.js:746]: good call, let me look

File: src/js/components/Layer/doc.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Layer/doc.js:161]: k

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**